app/server/src/main/scala/org/bitcoins/server/BitcoinSRunner.scala

```diff
@@ -80,7 +80,7 @@ trait BitcoinSRunner extends BitcoinSLogger {
   private lazy val network: BitcoinNetwork =
     BitcoinNetworks.fromString(networkStr)
 
-  lazy val datadir: Path = {
+  val datadir: Path = {
```

Comment:
I think we should keep this lazy. Since we don't actually have test cases for server startup (#2161 ). We have no way to detect regressions on CI. 

Also why is logging set here and not somewhere else? Doesn't this need to be done inside every submodule? If not, why isn't logging working as expected in the server project? It seems we have disconnect from library behavior and standalone binary behavior ?

Reply:
The log file location needs to be set here. We need to parse the config before so we know our datadir and network.

Reply:
>The log file location needs to be set here.  

Why? It seems odd to me that the logging property is set within `val datadir` to begin with.

Reply:
That is the soonest time we know it.

Reply:
Here is the scala documentation on this situation: https://docs.scala-lang.org/tutorials/FAQ/initialization-order.html

TLDR: Use `lazy val`

I was actually unaware of this compiler flag: `-Xcheckinit`. Thats pretty cool.

Reply:
I guess we can move it back to `lazy val`, we just then lose our guarantee that the log file location is set before the application starts

Reply:
Changed it so it uses a `lazy val` again and will now set the log location automatically before the setting up anything else